### PR TITLE
(checkers): Add flymake-collection-flake8

### PR DESCRIPTION
**Pull request by fbergroth**
Adding a checker that I was missing. :-)

### CODE REVIEW

File: tests/checkers/test-cases/pylint.yml [hunk 1]
```diff
@@ -26,4 +26,4 @@ tests:
     lints:
       - point: [1, 11]
         level: error
-        message: E0001 invalid syntax (<unknown>, line 1) (pylint)
+        message: "E0001 Parsing failed: 'invalid syntax (<unknown>, line 1)' (pylint)"
```
Comment (mohkale) [inline tests/checkers/test-cases/pylint.yml:29]: Nice, thanks for looking into this as well.

**Review by mohkale — APPROVED** [review]

**Comment by fbergroth** [comment]
[flycheck sets severity depending on the id](https://github.com/flycheck/flycheck/blob/b8f5bad487dafb942b577b571786e4495bd5a400/flycheck.el#L10578-L10607), which is somewhat nicer.

Also, perhaps there should be a defcustom for any extra args a user may want to pass?

**Comment by mohkale** [comment]
@fbergroth 

> [flycheck sets severity depending on the id](https://github.com/flycheck/flycheck/blob/b8f5bad487dafb942b577b571786e4495bd5a400/flycheck.el#L10578-L10607), which is somewhat nicer.

This can be supported here as well. I'd recommend defining a different regexp for the different IDs. See [rubocop](https://github.com/mohkale/flymake-collection/blob/cd2574574bda1f3d94007cdc3b7e0f8007d69716/src/checkers/flymake-collection-rubocop.el#L75) for an example.

> Also, perhaps there should be a defcustom for any extra args a user may want to pass?

I don't see any reason not to. Normally I'm fine waiting until someone asks for this than adding it on for each new checker. It's just different commands have different conventions, some should work reasonably well out of the box or load an apropriate convention from the local project. Trying to tailor to each individual use case or make it so flexible conflicting use cases can also arise is IMO too much burden when just adding a new checker. That said if you feel it's worth adding now I'd be happy to accept. More customisability better than less :-).

**Comment by fbergroth** [comment]
> This can be supported here as well. I'd recommend defining a different regexp for the different IDs. See [rubocop](https://github.com/mohkale/flymake-collection/blob/cd2574574bda1f3d94007cdc3b7e0f8007d69716/src/checkers/flymake-collection-rubocop.el#L75) for an example.

Yes. The problem is that they get hardcoded, and flake8 has a ton of plugins, which adds maintenance burden. Perhaps it's reasonable to try match the errors, and let the rest be warnings. But I think it's fine to let it be warnings for now.

> I don't see any reason not to. Normally I'm fine waiting until someone asks for this than adding it on for each new checker. It's just different commands have different conventions, some should work reasonably well out of the box or load an apropriate convention from the local project. Trying to tailor to each individual use case or make it so flexible conflicting use cases can also arise is IMO too much burden when just adding a new checker. That said if you feel it's worth adding now I'd be happy to accept. More customisability better than less :-).

Cool, I added it, and also --stdin-display-name. This is done, unless you want any further changes?

**Review by mohkale — APPROVED** [review]

**Pull request merged**